common/djangoapps/course_modes/views.py

```diff
@@ -123,15 +126,21 @@ def get(self, request, course_id, error=None):
             "nav_hidden": True,
         }
         if "verified" in modes:
+            verified_mode = modes["verified"]
             context["suggested_prices"] = [
                 decimal.Decimal(x.strip())
-                for x in modes["verified"].suggested_prices.split(",")
+                for x in verified_mode.suggested_prices.split(",")
                 if x.strip()
             ]
-            context["currency"] = modes["verified"].currency.upper()
-            context["min_price"] = modes["verified"].min_price
-            context["verified_name"] = modes["verified"].name
-            context["verified_description"] = modes["verified"].description
+            context["currency"] = verified_mode.currency.upper()
+            context["min_price"] = verified_mode.min_price
+            context["verified_name"] = verified_mode.name
+            context["verified_description"] = verified_mode.description
+
+            if verified_mode.sku:
+                context["use_ecommerce_payment_flow"] = waffle.flag_is_active(request, 'use-ecommerce-payment-flow')
+                context["ecommerce_payment_page"] = urljoin(settings.ECOMMERCE_PUBLIC_URL_ROOT, settings.ECOMMERCE_COURSE_CHECKOUT_PATH)
```

Comment:
Something that Miki suggested, and that I think makes sense, was to adopt a standard pattern of using a single redirect url for links between our services. Basically, a URL that functions as the equivalent to the django `reverse` function. It would mean that the separate services would be freer to adjust their urls to optimize user experience, and would provide "keyword arguments" when passing between services.

Given that, this might look something like: `urljoin(settings.ECOMMERCE_PUBLIC_URL_ROOT, 'goto') + '?action=pay&sku={sku}'`